people-and-planet-ai/land-cover-classification/serving/data.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+from typing import Tuple
+
+import ee
+from google.api_core import exceptions, retry
+import google.auth
+import numpy as np
+from numpy.lib.recfunctions import structured_to_unstructured
+import requests
+
+
+SCALE = 10  # meters per pixel
+
+
+def ee_init() -> None:
+    """Authenticate and initialize Earth Engine with the default credentials."""
+    # Use the Earth Engine High Volume endpoint.
+    #   https://developers.google.com/earth-engine/cloud/highvolume
+    credentials, project = google.auth.default(
+        scopes=[
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/earthengine",
+        ]
+    )
+    ee.Initialize(
+        credentials.with_quota_project(None),
+        project=project,
+        opt_url="https://earthengine-highvolume.googleapis.com",
+    )
+
+
+def get_input_image(year: int) -> ee.Image:
+    """Get a Sentinel-2 Earth Engine image with elevation.
+
+    This filters clouds and returns the median for the selected time range.
+
+    For more information, see:
+        https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_S2_HARMONIZED
+        https://developers.google.com/earth-engine/datasets/catalog/CGIAR_SRTM90_V4
+
+    Args:
+        start_date: Beginning of the time range to consider for the median.
```

Comment:
Docstrings seem to be inconsistent with `(year: int)`

Reply:
Looks like this function could raise Exceptions. Consider documenting (optional).